lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -2307,6 +2307,11 @@ namespace Js
 
     BOOL JavascriptOperators::SetProperty(Var receiver, RecyclableObject* object, PropertyId propertyId, Var newValue, PropertyValueInfo * info, ScriptContext* requestContext, PropertyOperationFlags propertyOperationFlags)
     {
+        if (propertyId == PropertyIds::next && object == requestContext->GetLibrary()->GetArrayIteratorPrototype())
+        {
+            requestContext->GetLibrary()->SetArrayIteratorHasUserDefinedNext(true);
+        }
+
```

Comment:
I agree with @ianwjhalliday here. Overriding the next here won't work with the inlinecache. There are too many SetProperty overrides which can land in next being added to the prototype object.  We should find and cost the correct solution. To take it for 1.2 depends on the cost of the solution. 

---
In reply to: [57463986](https://github.com/Microsoft/ChakraCore/pull/635#discussion_r57463986) [](ancestors = 57463986)